base/multidimensional.jl

```diff
@@ -1534,6 +1534,9 @@ end
     end
 end
 
+isassigned(a::AbstractArray, i::CartesianIndex) = isassigned(a, Tuple(i)...)
+isassigned(a::AbstractArray, i::Union{Int, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```

Comment:
```suggestion
isassigned(a::AbstractArray, i::Union{Integer, CartesianIndex}...) = isassigned(a, CartesianIndex(i))
```
This fix the stack overflow locally.

Reply:
This stack overflow should be predictable. As the general fallback is defined as:
```julia
function isassigned(a::AbstractArray, i::Integer...)
```
Thus we encounter a circular subtype chain.
Have no idea on the test though.

Reply:
you said "locally", just put that into tests?

Reply:
Any `AbstractArray` using try based fallback should be testable. e.g. `isassigned(view([1],1:1),1)`.
But this should be tested implicitly via `show`. (unassigned value is printed as #undef in REPL)

Reply:
alright, no tests then, testing for not stack overflowing is weird anyway